src/Compilers/CSharp/Test/Emit/CodeGen/IndexAndRangeTests.cs

```diff
@@ -17,6 +18,636 @@ private CompilationVerifier CompileAndVerifyWithIndexAndRange(string s, string e
+  IL_0012:  call       ""S..ctor(int[])""
+  IL_0017:  ldloca.s   V_1
+  IL_0019:  stloc.2
+  IL_001a:  ldloc.2
+  IL_001b:  call       ""int S.Length.get""
+  IL_0020:  ldc.i4.1
+  IL_0021:  sub
+  IL_0022:  ldloc.2
+  IL_0023:  stloc.3
+  IL_0024:  stloc.s    V_4
+  IL_0026:  ldloc.3
+  IL_0027:  ldloc.s    V_4
+  IL_0029:  ldloc.3
+  IL_002a:  ldloc.s    V_4
+  IL_002c:  call       ""int S.this[int].get""
+  IL_0031:  ldc.i4.5
+  IL_0032:  add
+  IL_0033:  call       ""void S.this[int].set""
+  IL_0038:  ldloc.0
+  IL_0039:  ldc.i4.1
+  IL_003a:  ldelem.i4
+  IL_003b:  call       ""void System.Console.WriteLine(int)""
+  IL_0040:  ret
+}");
+        }
+
+        [Fact]
+        [WorkItem(37789, "https://github.com/dotnet/roslyn/issues/37789")]
+        public void PatternRangeCompoundOperator()
+        {
+            var src = @"
+using System;
+struct S
+{
+    private readonly int[] _array;
+
+    private int _counter;
+
+    public S(int[] a)
+    {
+        _array = a;
+        _counter = 0;
+    }
+    public int Length
+    {
+        get
+        {
+            Console.WriteLine(""Length "" + _counter++);
+            return _array.Length;
+        }
+    }
+    public ref int Slice(int start, int length)
+    {
+        Console.WriteLine(""Slice "" + _counter++);
+        return ref _array[start];
+    }
+}
+class C
+{
+    static void Main(string[] args)
+    {
+        var array = new int[2];
+        Console.WriteLine(array[1]);
+        var s = new S(array);
+        s[1..] += 5;
+        Console.WriteLine(array[1]);
+
+    }
+}
+";
+            var comp = CreateCompilationWithIndexAndRangeAndSpan(src, TestOptions.ReleaseExe);
+            var verifier = CompileAndVerify(comp, expectedOutput: @"0
+Length 0
+Slice 1
+5");
+            verifier.VerifyIL("C.Main", @"
+{
+  // Code size       57 (0x39)
+  .maxstack  3
+  .locals init (int[] V_0, //array
+                S V_1,
+                int V_2,
+                int V_3)
+  IL_0000:  ldc.i4.2
+  IL_0001:  newarr     ""int""
+  IL_0006:  stloc.0
+  IL_0007:  ldloc.0
+  IL_0008:  ldc.i4.1
+  IL_0009:  ldelem.i4
+  IL_000a:  call       ""void System.Console.WriteLine(int)""
+  IL_000f:  ldloc.0
+  IL_0010:  newobj     ""S..ctor(int[])""
+  IL_0015:  stloc.1
+  IL_0016:  ldloca.s   V_1
+  IL_0018:  call       ""int S.Length.get""
+  IL_001d:  ldc.i4.1
+  IL_001e:  stloc.2
+  IL_001f:  ldloc.2
+  IL_0020:  sub
+  IL_0021:  stloc.3
+  IL_0022:  ldloca.s   V_1
+  IL_0024:  ldloc.2
+  IL_0025:  ldloc.3
+  IL_0026:  call       ""ref int S.Slice(int, int)""
+  IL_002b:  dup
+  IL_002c:  ldind.i4
+  IL_002d:  ldc.i4.5
+  IL_002e:  add
+  IL_002f:  stind.i4
+  IL_0030:  ldloc.0
+  IL_0031:  ldc.i4.1
+  IL_0032:  ldelem.i4
+  IL_0033:  call       ""void System.Console.WriteLine(int)""
+  IL_0038:  ret
+}");
+        }
+
+        [Fact]
+        [WorkItem(37789, "https://github.com/dotnet/roslyn/issues/37789")]
+        public void PatternindexNullableCoalescingAssignmentClass()
```

Comment:
>index [](start = 27, length = 5)

"Index" (uppercase "I") here and below.

Reply:
Yuck, I'd like to fix it, but not worth resetting the CI at this point.